src/evaluators/bc/PHAL_Dirichlet_Def.hpp

```diff
@@ -26,9 +27,26 @@ DirichletBase(Teuchos::ParameterList& p) :
   value = p.get<RealType>("Dirichlet Value");
 
   std::string name = p.get< std::string >("Dirichlet Name");
+  isRandom = false;
+  if (p.isParameter("Random"))
+    isRandom = p.get< bool >("Random");
   const Teuchos::RCP<PHX::DataLayout> dummy = p.get< Teuchos::RCP<PHX::DataLayout> >("Data Layout");
   const PHX::Tag<ScalarT> fieldTag(name, dummy);
 
+  if(isRandom) {
```

Comment:
This is a bit of a hack. Is there a way we can move this code out of this evaluator?

Reply:
I agree. But I think the only alternative is to create a new Dirichlet ev type, something like `DirichletField`, but that already exist (though we could rename that one to `DirichletDistributedField`). Then, modify the Dirichlet factory, so that it can build the SharedParam and RandomPhysicalParam evaluators. That's however hard to do with current yaml syntax. We'd need something like
```
DBC on NodeSet blah for DOF XYZ prescribe Field: [random_phys_name, shared_param_name]
```
so that the factory can auto build the SharedParam ev for $shared_param_name, and the RandomPhysicalParameter for $random_phys_name...

P.s: I've been willing to re-do how our DBC/NBC are prescribed, by having sublists, rather than encoding bc type, ns name, and dof name all in the yaml parameter entry... With something like
```
   DBC:
      Number DBCs: 2
      DBC 0:
         Type: prescribe field
         DOF: xyz
         Node Set: blah
         Field Name: beta
      DOF 1:
         Type: Value
          DOF: xyz
          Node Set: blah2
          Value: 0.0
```
It would be less hacky to implement this new DBC as a new evaluator.

Reply:
I like this idea. I can look into that but I will not be able to do so before my holiday.
If preferred, we can leave the PR open and I can do that inside this PR.

Reply:
I don't like PRs open for too long. If you plan to do the change, you can leave things as they are, and do a follow up PR soon.

Reply:
Agreed

Reply:
OK, I will work on that in another PR then asap but probably after Christmas. Let me know if you want any other modifications in this PR.